### PR TITLE
[GRANT-172]  - Exception Handling

### PR DESCRIPTION
**Pull request by leon-fw**
Added some basic custom errors for Application & User.

### CODE REVIEW

File: server/src/application/application.errors.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+export const ApplicationError = {
+  APPLICATION_NOT_FOUND: {
+    errorType: 'APPLICATION_NOT_FOUND',
+    errorMessage: 'Cannot find application',
+    httpStatus: HttpStatus.NOT_FOUND,
+  } as GenericError,
+
+  USER_EMPTY: {
+    errorType: 'USER_EMPTY',
```
Comment (karthikktamilmani-freshworks) [inline server/src/application/application.errors.ts:12]: User not found

File: server/src/application/application.errors.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    errorMessage: 'Empty User is not allowed',
+    httpStatus: HttpStatus.BAD_REQUEST,
```
Comment (karthikktamilmani-freshworks) [inline server/src/application/application.errors.ts:14]: Not found

File: server/src/application/application.service.ts [hunk 3]
```diff
@@ -51,19 +53,18 @@ export class ApplicationService {
     query.filter(queryBuilder);
 
     const applications = await queryBuilder.getManyAndCount();
+
     return new PaginationRO<Application>(applications);
   }
 
   async getApplication(applicationId: string): Promise<Application | undefined> {
```
Comment (karthikktamilmani-freshworks) [inline server/src/application/application.service.ts:60]: Will always return Promise<Application>

File: server/src/application/application.service.ts [hunk 4]
```diff
@@ -87,21 +88,17 @@ export class ApplicationService {
     const application = await this.getApplication(applicationId);
     if (application) {
```
Comment (karthikktamilmani-freshworks) [inline server/src/application/application.service.ts:89]: this can be removed!

File: server/src/application/application.service.ts [hunk 5]
```diff
@@ -115,9 +112,9 @@ export class ApplicationService {
 
   async createComment(applicationId: string, commentDto: CommentDto, user: User): Promise<Comment> {
     const application = await this.getApplication(applicationId);
-    if (application && user) {
-      return await this.commentService.createComment(commentDto, application, user);
+    if (!user) {
```
Comment (karthikktamilmani-freshworks) [inline server/src/application/application.service.ts:115]: This scenario will not exist

**Review by karthikktamilmani-freshworks — CHANGES_REQUESTED** [review]

File: server/src/application/application.service.ts (hunk 5 above)
Comment (leon-fw) [reply server/src/application/application.service.ts:115]: Thank you, was not sure about this!

File: server/src/application/application.errors.ts (hunk 1 above)
Comment (leon-fw) [reply server/src/application/application.errors.ts:12]: This one is not used anywhere anymore, so removing

File: server/src/application/application.errors.ts (hunk 2 above)
Comment (leon-fw) [reply server/src/application/application.errors.ts:14]: This one is not used anywhere anymore, so removing

**Review by karthikktamilmani-freshworks — APPROVED** [review]
Great!

**Pull request merged**